File: test_complete_flow.py
import requests
import json
import sqlite3
from datetime import datetime

def test_complete_flow():
    """
    Probar el flujo completo: Frontend -> Backend -> Base de datos con auto-incremento
    """
    
    # URL del backend
    base_url = "http://localhost:8080/api/formulario"
    
    print("=== PRUEBA COMPLETA DEL FLUJO ===")
    
    # 1. Preparar datos de prueba
    usuario_test = {
        "nombre": "Usuario Prueba Completa",
        "cedula": 123456789,
        "correo": "usuario.prueba@test.com",
        "numeroCelular": 3001234567,
        "numeroFijo": 6012345678,
        "cedulaExpedicion": "Bogotá",
        "paisNacimiento": "Colombia",
        "ciudadNacimiento": "Bogotá",
        "cargo": "Desarrollador",
        "area": "Tecnología",
        "estadoCivil": "Soltero",
        "tipoSangre": "O+",
        "fechaNacimiento": "1990-05-15"
    }
    
    print(f"\n📋 Datos de prueba preparados:")
    print(f"  Nombre: {usuario_test['nombre']}")
    print(f"  Cédula: {usuario_test['cedula']}")
    print(f"  Email: {usuario_test['correo']}")
    
    try:
        # 2. Verificar que el backend esté ejecutándose
        print(f"\n🔗 Verificando conexión con el backend...")
        try:
            response = requests.get(f"{base_url}/estado/123456789", timeout=5)
            print(f"✅ Backend está ejecutándose")
        except requests.exceptions.RequestException as e:
            print(f"❌ Backend no está disponible: {e}")
            print(f"💡 Asegúrate de que el servidor Spring Boot esté ejecutándose en puerto 8080")
            return
        
        # 3. Enviar datos al endpoint de información personal
        print(f"\n📤 Enviando datos al backend...")
        headers = {
            'Content-Type': 'application/json',
            'Accept': 'application/json'
        }
        
        response = requests.post(
            f"{base_url}/informacion-personal/guardar",
            json=usuario_test,
            headers=headers,
            timeout=10
        )
        
        print(f"📊 Respuesta del backend:")
        print(f"  Status Code: {response.status_code}")
        print(f"  Headers: {dict(response.headers)}")
        
        if response.status_code == 200:
            response_data = response.json()
            print(f"  Response JSON: {response_data}")
            
            if response_data.get('success'):
                usuario_guardado = response_data.get('data', {})
                id_generado = usuario_guardado.get('id')
                
                if id_generado:
                    print(f"✅ Usuario guardado exitosamente con ID: {id_generado}")
                    
                    # 4. Verificar en la base de datos
                    print(f"\n🔍 Verificando en la base de datos...")
                    conn = sqlite3.connect('bd/bd.db')
                    cursor = conn.cursor()
                    
                    cursor.execute("""
                        SELECT ID_USUARIO, DOCUMENTO, NOMBRE, CORREO 
                        FROM USUARIO 
                        WHERE ID_USUARIO = ?
                    """, (id_generado,))
                    
                    usuario_db = cursor.fetchone()
                    
                    if usuario_db:
                        print(f"✅ Usuario encontrado en BD:")
                        print(f"  ID: {usuario_db[0]}")
                        print(f"  Cédula: {usuario_db[1]}")
                        print(f"  Nombre: {usuario_db[2]}")
                        print(f"  Email: {usuario_db[3]}")
                        
                        # 5. Limpiar usuario de prueba
                        cursor.execute("DELETE FROM USUARIO WHERE ID_USUARIO = ?", (id_generado,))
                        conn.commit()
                        print(f"\n🧹 Usuario de prueba eliminado de la BD")
                        
                    else:
                        print(f"❌ Usuario no encontrado en la BD")
                    
                    conn.close()
                    
                else:
                    print(f"❌ No se retornó ID del usuario guardado")
            else:
                print(f"❌ Error en el backend: {response_data.get('message')}")
        else:
            print(f"❌ Error HTTP {response.status_code}: {response.text}")
        
        print(f"\n✅ Prueba completada")
        
    except Exception as e:
        print(f"❌ Error durante la prueba: {e}")

if __name__ == "__main__":
    test_complete_flow() 

File: test_complete_flow.js
const axios = require('axios');

const BASE_URL = 'http://localhost:8080';
const TOKEN = 'test-token';

// Configurar axios
const api = axios.create({
  baseURL: BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${TOKEN}`
  }
});

async function testCompleteFlow() {
  try {
    console.log('🚀 Iniciando prueba del flujo completo...\n');

    // Paso 1: Guardar información personal (crear nuevo usuario)
    console.log('📝 Paso 1: Guardando información personal...');
    const personalData = {
      documento: 1006101211,
      cedulaExpedicion: "Cali",
      nombre: "JESUS FELIPE CORDOBA ECHANDIA",
      fechaNacimiento: "2001-10-25",
      paisNacimiento: "Colombia",
      ciudadNacimiento: "Cali",
      cargo: "APRENDIZ TIC SENIOR",
      area: "SOLUCIONES INFORMATICAS",
      estadoCivil: "Soltero",
      tipoSangre: "O+",
      numeroCelular: 3186183326,
      numeroFijo: 3104128521,
      numeroCorp: 3186183326,
      correo: "jfcordoba@gana.com.co"
    };

    const personalResponse = await api.post('/api/formulario/informacion-personal/guardar', personalData);
    console.log('✅ Información personal guardada:', personalResponse.data);

    // Obtener el ID del usuario de la respuesta
    const userId = personalResponse.data.data?.idUsuario || personalResponse.data.data?.id;
    console.log('👤 ID de usuario obtenido:', userId);

    if (!userId) {
      console.error('❌ No se pudo obtener el ID del usuario');
      return;
    }

    // Paso 2: Guardar información académica
    console.log('\n📚 Paso 2: Guardando información académica...');
    const academicData = [
      {
        nivelAcademico: "Universidad",
        programa: "Ingenieria de Sistemas",
        institucion: "Universidad Nacional",
        semestre: 10,
        graduacion: "Graduado"
      }
    ];

    const academicResponse = await api.post(`/api/formulario/estudios/guardar?idUsuario=${userId}`, academicData);
    console.log('✅ Información académica guardada:', academicResponse.data);

    // Paso 3: Guardar información de vehículos
    console.log('\n🚗 Paso 3: Guardando información de vehículos...');
    const vehicleData = [
      {
        marca: "Toyota",
        modelo: "Corolla",
        placa: "ABC123",
        tipo: "PARTICULAR",
        color: "Blanco",
        anio: 2020
      }
    ];

    const vehicleResponse = await api.post(`/api/formulario/vehiculos/guardar?idUsuario=${userId}`, vehicleData);
    console.log('✅ Información de vehículos guardada:', vehicleResponse.data);

    // Paso 4: Guardar información de vivienda
    console.log('\n🏠 Paso 4: Guardando información de vivienda...');
    const housingData = {
      direccion: "Calle 123 #45-67",
      tipoVivienda: "Casa",
      barrio: "Centro",
      ciudad: "Cali",
      departamento: "Valle del Cauca"
    };

    const housingResponse = await api.post(`/api/formulario/vivienda/guardar?idUsuario=${userId}`, housingData);
    console.log('✅ Información de vivienda guardada:', housingResponse.data);

    // Paso 5: Guardar personas a cargo
    console.log('\n👨‍👩‍👧‍👦 Paso 5: Guardando personas a cargo...');
    const dependentsData = [
      {
        nombre: "María González",
        parentesco: "Madre",
        fechaNacimiento: "1975-05-15",
        ocupacion: "Ama de casa",
        telefono: "3001234567"
      }
    ];

    const dependentsResponse = await api.post(`/api/formulario/personas-acargo/guardar?idUsuario=${userId}`, dependentsData);
    console.log('✅ Personas a cargo guardadas:', dependentsResponse.data);

    // Paso 6: Guardar contactos de emergencia
    console.log('\n📞 Paso 6: Guardando contactos de emergencia...');
    const contactsData = [
      {
        nombre: "Carlos Rodríguez",
        parentesco: "Hermano",
        telefono: "3109876543",
        direccion: "Calle 456 #78-90",
        ocupacion: "Ingeniero"
      }
    ];

    const contactsResponse = await api.post(`/api/formulario/contactos/guardar?idUsuario=${userId}`, contactsData);
    console.log('✅ Contactos de emergencia guardados:', contactsResponse.data);

    // Paso 7: Guardar declaraciones de conflicto
    console.log('\n⚖️ Paso 7: Guardando declaraciones de conflicto...');
    const declarationsData = [
      {
        tipoRelacion: "Familiar",
        nombrePersona: "Ana Martínez",
        cargo: "Gerente",
        empresa: "Empresa ABC",
        descripcion: "Hermana trabaja en empresa relacionada"
      }
    ];

    const declarationsResponse = await api.post(`/api/formulario/relaciones-conflicto/guardar?idUsuario=${userId}`, declarationsData);
    console.log('✅ Declaraciones de conflicto guardadas:', declarationsResponse.data);

    console.log('\n🎉 ¡Flujo completo probado exitosamente!');
    console.log(`📊 Usuario ID: ${userId}`);
    console.log('✅ Todos los pasos se guardaron correctamente con auditoría');

  } catch (error) {
    console.error('❌ Error en el flujo completo:', error.response?.data || error.message);
  }
}

// Ejecutar la prueba
testCompleteFlow(); 